src/battle_util.c

```diff
@@ -2084,6 +2085,24 @@ void TryToRevertMimicry(void)
     }
 }
 
+u32 GetMonFriendshipScore(struct Pokemon *pokemon)
+{
```

Comment:
You could have a local variable there as to not call the function 7 times in a row.